packages/pdf-viewer/hooks/usePdfData.ts

```diff
@@ -0,0 +1,19 @@
+import { useState } from 'react';
+import useAsyncEffect, { AsyncEffectEvent } from '@joplin/lib/hooks/useAsyncEffect';
+import { PdfData } from '../pdfSource';
+
+
+const usePdfData = (pdfPath: string) => {
+	const [pdf, setPdf] = useState<PdfData>(null);
+
+	useAsyncEffect(async (event: AsyncEffectEvent) => {
+		const pdfData = new PdfData();
+		await pdfData.loadDoc(pdfPath);
+		if (event.cancelled) return;
+		setPdf(pdfData);
+	}, []);
```

Comment:
You don't get how hooks work - please read the doc. pdfPath is an obvious dependency.